datumaro/components/converter.py

```diff
@@ -76,15 +85,26 @@ def _save_image(self, item, path=None, *,
             self._make_image_filename(item, name=name, subdir=subdir))
         path = osp.abspath(path)
 
-        src_ext = item.image.ext.lower()
-        dst_ext = osp.splitext(osp.basename(path))[1].lower()
+        item.image.save(path)
+
+    def _save_pcd(self, item=None, path=None, *,
+            name=None, subdir=None, basedir=None):
+        assert not ((subdir or name or basedir) and path), \
+            "Can't use both subdir or name or basedir and path arguments"
+
+        if not item.pcd:
+            log.warning("Item '%s' has no pcd", item.id)
+            return
+
+        basedir = basedir or self._save_dir
+        path = path or osp.join(basedir,
+            self._make_pcd_filename(item, name=name, subdir=subdir))
+        path = osp.abspath(path)
 
         os.makedirs(osp.dirname(path), exist_ok=True)
-        if src_ext == dst_ext and osp.isfile(item.image.path):
-            if item.image.path != path:
-                shutil.copyfile(item.image.path, path)
-        elif src_ext == dst_ext and isinstance(item.image, ByteImage):
+        if item.pcd and osp.isfile(item.pcd):
+            if item.pcd != path:
+                shutil.copyfile(item.pcd, path)
+        elif isinstance(item.pcd, bytes):
```

Comment:
This branch should be first, because otherwise it's possible that the `bytes` value gets interpreted as a path instead (very unlikely, but still).

Reply:
Probably, we need to delete it at all.